Test flake in querier.TestChunkQueryable

I see this failure in CI:
```
panic: test timed out after 1m0s

goroutine 152 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0x146
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x52

goroutine 1 [chan receive, 1 minutes]:
testing.(*T).Run(0xc4200d40f0, 0x1e2f2b4, 0x12, 0x1e9ae78, 0xc4208c7c01)
	/usr/local/go/src/testing/testing.go:825 +0x597
testing.runTests.func1(0xc4200d40f0)
	/usr/local/go/src/testing/testing.go:1063 +0xa5
testing.tRunner(0xc4200d40f0, 0xc4208c7d48)
	/usr/local/go/src/testing/testing.go:777 +0x16e
testing.runTests(0xc42000c400, 0x2a8b160, 0x7, 0x7, 0xc420138080)
	/usr/local/go/src/testing/testing.go:1061 +0x4e2
testing.(*M).Run(0xc420138080, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x2ce
main.main()
	_testmain.go:274 +0x335

goroutine 50 [syscall, 1 minutes]:
os/signal.signal_recv(0x488b61)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 12 [select]:
github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc420119400)
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x294
created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x205

goroutine 105 [chan receive]:
testing.(*T).Run(0xc42001c000, 0xc4206422a0, 0x1e, 0xc4223b49c0, 0x3)
	/usr/local/go/src/testing/testing.go:825 +0x597
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable(0xc42001c000)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:85 +0xb7
testing.tRunner(0xc42001c000, 0x1e9ae78)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 150 [runnable]:
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.funcRate(0xc4223d6da0, 0x1, 0x1, 0xc4225a4000, 0x1, 0x1, 0xc4206e2300, 0xc420396400, 0x5, 0x10)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/functions.go:147 +0xee
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).eval(0xc4208d05c0, 0x2051780, 0xc420358000, 0xc420c67858, 0xc4208d05c0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:877 +0x1bf8
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).Eval(0xc4208d05c0, 0x2051780, 0xc420358000, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:614 +0x8e
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).execEvalStmt(0xc4205e0510, 0x205ba40, 0xc4206e2060, 0xc4206e2000, 0xc42025e320, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:426 +0x696
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).exec(0xc4205e0510, 0x205ba40, 0xc4206e2060, 0xc4206e2000, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:340 +0x5a8
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*query).Exec(0xc4206e2000, 0x205ba00, 0xc4200480d0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:142 +0xde
github.com/weaveworks/cortex/pkg/querier.testQuery(0x204aaa0, 0xc42001c1e0, 0x2043fc0, 0xc42064ed60, 0x9a7ec800, 0x1e29c43, 0xd, 0x2aba500, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:140 +0x2bf
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable.func1(0xc42001c1e0)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:88 +0x1af
testing.tRunner(0xc42001c1e0, 0xc4223b49c0)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 114 [chan receive]:
github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2a99b40)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0xac
created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.0
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x231
FAIL	github.com/weaveworks/cortex/pkg/querier	60.902s
```

https://circleci.com/gh/grafana/cortex/957
```
panic: test timed out after 1m0s

goroutine 56 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0x146
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x52

goroutine 1 [chan receive, 1 minutes]:
testing.(*T).Run(0xc4202740f0, 0x1e2d233, 0x12, 0x1e98e50, 0xc420897c01)
	/usr/local/go/src/testing/testing.go:825 +0x597
testing.runTests.func1(0xc4202740f0)
	/usr/local/go/src/testing/testing.go:1063 +0xa5
testing.tRunner(0xc4202740f0, 0xc420897d48)
	/usr/local/go/src/testing/testing.go:777 +0x16e
testing.runTests(0xc420746640, 0x2a88660, 0x7, 0x7, 0xc42062a080)
	/usr/local/go/src/testing/testing.go:1061 +0x4e2
testing.(*M).Run(0xc42062a080, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x2ce
main.main()
	_testmain.go:282 +0x335

goroutine 37 [syscall, 1 minutes]:
os/signal.signal_recv(0x488b61)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 9 [select]:
github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc4203ba000)
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x294
created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x205

goroutine 76 [chan receive]:
github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2a97040)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0xac
created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.0
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x231

goroutine 165 [chan receive]:
testing.(*T).Run(0xc420274780, 0xc42030c1e0, 0x1e, 0xc420746000, 0x3)
	/usr/local/go/src/testing/testing.go:825 +0x597
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable(0xc420274780)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:85 +0xb7
testing.tRunner(0xc420274780, 0x1e98e50)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 157 [runnable]:
github.com/weaveworks/cortex/pkg/querier.(*nonOverlappingIterator).AtTime(0xc421232fe0, 0x1)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_merge_iterator.go:200 +0xda
github.com/weaveworks/cortex/pkg/querier.(*seriesIteratorHeap).Less(0xc4207e8608, 0x1, 0x0, 0x1ad2200)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_merge_iterator.go:146 +0xb7
container/heap.down(0x205dc20, 0xc4207e8608, 0x0, 0x4, 0xc421233008)
	/usr/local/go/src/container/heap/heap.go:114 +0xad
container/heap.Fix(0x205dc20, 0xc4207e8608, 0x0)
	/usr/local/go/src/container/heap/heap.go:87 +0x6f
github.com/weaveworks/cortex/pkg/querier.(*chunkMergeIterator).Next(0xc4207e85f0, 0x91c67f28)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_merge_iterator.go:113 +0x245
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage.(*BufferedSeriesIterator).Next(0xc4205c57d0, 0xc420c48f01)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage/buffer.go:118 +0xea
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage.(*BufferedSeriesIterator).Seek(0xc4205c57d0, 0x91c6b9c0, 0x10)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage/buffer.go:100 +0xc4
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).matrixIterSlice(0xc42003a900, 0xc4205c57d0, 0x91c5cf61, 0x91c6b9c0, 0xc42000a700, 0x1, 0x10, 0xc4205c5770, 0x1, 0x1)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:1104 +0x1f6
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).eval(0xc42003a900, 0x204f620, 0xc42121b2a0, 0xc420c49858, 0xc42003a900)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:870 +0x1a03
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).Eval(0xc42003a900, 0x204f620, 0xc42121b2a0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:614 +0x8e
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).execEvalStmt(0xc4205c51a0, 0x20598e0, 0xc4208be300, 0xc4208be2a0, 0xc4207e80a0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:426 +0x696
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).exec(0xc4205c51a0, 0x20598e0, 0xc4208be300, 0xc4208be2a0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:340 +0x5a8
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*query).Exec(0xc4208be2a0, 0x20598a0, 0xc42010a050, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:142 +0xde
github.com/weaveworks/cortex/pkg/querier.testQuery(0x2048900, 0xc4202741e0, 0x2041e20, 0xc42121aec0, 0x9a7ec800, 0x1e27bf7, 0xd, 0x2ab7a00, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:140 +0x2bf
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable.func1(0xc4202741e0)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:88 +0x1af
testing.tRunner(0xc4202741e0, 0xc420746000)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565
FAIL	github.com/weaveworks/cortex/pkg/querier	60.776s
```

https://circleci.com/gh/grafana/cortex/984
```
panic: test timed out after 1m0s

goroutine 56 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0x146
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x52

goroutine 1 [chan receive, 1 minutes]:
testing.(*T).Run(0xc420570000, 0x1ab355d, 0x12, 0x1b344b0, 0xc42056dc01)
	/usr/local/go/src/testing/testing.go:825 +0x597
testing.runTests.func1(0xc420570000)
	/usr/local/go/src/testing/testing.go:1063 +0xa5
testing.tRunner(0xc420570000, 0xc42056dd48)
	/usr/local/go/src/testing/testing.go:777 +0x16e
testing.runTests(0xc420120a40, 0x2544260, 0x8, 0x8, 0xc4202ac100)
	/usr/local/go/src/testing/testing.go:1061 +0x4e2
testing.(*M).Run(0xc4202ac100, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x2ce
main.main()
	_testmain.go:294 +0x335

goroutine 5 [syscall, 1 minutes]:
os/signal.signal_recv(0x488c11)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 51 [select]:
github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc4203aa000)
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x294
created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x205

goroutine 96 [chan receive]:
github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2552a60)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0xac
created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.0
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x231

goroutine 135 [runnable]:
github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk.(*varbitChunk).addLaterSample(0xc4207ed500, 0x70d85228, 0x41dc36148a000000, 0x430235, 0x194a860, 0x1a6aaa0, 0x1a6aa00, 0x1c65660, 0xc4200b3b38)
	/go/src/github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk/varbit.go:569 +0x23f
github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk.(*varbitChunk).Add(0xc4207ed500, 0x70d85228, 0x41dc36148a000000, 0xc4207e7020, 0x1, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk/varbit.go:275 +0x22d
github.com/weaveworks/cortex/pkg/querier.mkChunk(0x1c6eba0, 0xc4205700f0, 0x70a71c80, 0x714be800, 0x37e11d600, 0x202, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:123 +0x1ef
github.com/weaveworks/cortex/pkg/querier.makeMockChunkStore(0x1c6eba0, 0xc4205700f0, 0x2d0, 0xc42004e502, 0x1, 0xc420e3e030, 0x56d09b)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:109 +0x149
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable.func1(0xc4205700f0)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:86 +0xb7
testing.tRunner(0xc4205700f0, 0xc4206ec040)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 107 [chan receive]:
testing.(*T).Run(0xc420570690, 0xc4202945c0, 0x1e, 0xc4206ec040, 0x3)
	/usr/local/go/src/testing/testing.go:825 +0x597
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable(0xc420570690)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:85 +0xb7
testing.tRunner(0xc420570690, 0x1b344b0)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565
FAIL	github.com/weaveworks/cortex/pkg/querier	60.244s
```

https://circleci.com/gh/grafana/cortex/987
```
panic: test timed out after 1m0s

goroutine 162 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0x146
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x52

goroutine 1 [chan receive, 1 minutes]:
testing.(*T).Run(0xc4200ca0f0, 0x1e3618c, 0x12, 0x1ea1eb0, 0xc42086fc01)
	/usr/local/go/src/testing/testing.go:825 +0x597
testing.runTests.func1(0xc4200ca0f0)
	/usr/local/go/src/testing/testing.go:1063 +0xa5
testing.tRunner(0xc4200ca0f0, 0xc42086fd48)
	/usr/local/go/src/testing/testing.go:777 +0x16e
testing.runTests(0xc4205837a0, 0x2a92da0, 0x7, 0x7, 0xc42015e500)
	/usr/local/go/src/testing/testing.go:1061 +0x4e2
testing.(*M).Run(0xc42015e500, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x2ce
main.main()
	_testmain.go:274 +0x335

goroutine 23 [syscall, 1 minutes]:
os/signal.signal_recv(0x488b61)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 52 [select]:
github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc42039a000)
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x294
created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram
	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x205

goroutine 88 [chan receive]:
testing.(*T).Run(0xc420166000, 0xc4201ac1a0, 0x1e, 0xc4207b6520, 0x3)
	/usr/local/go/src/testing/testing.go:825 +0x597
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable(0xc420166000)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:85 +0xb7
testing.tRunner(0xc420166000, 0x1ea1eb0)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 97 [runnable]:
github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk.(*varbitChunkIterator).readDDT(0xc4226cb000)
	/go/src/github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk/varbit.go:1095 +0x3aa
github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk.(*varbitChunkIterator).Scan(0xc4226cb000, 0x1)
	/go/src/github.com/weaveworks/cortex/pkg/prom1/storage/local/chunk/varbit.go:1005 +0xa33
github.com/weaveworks/cortex/pkg/querier.(*chunkIterator).Next(0xc421437ea0, 0xc400000001)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_iterator.go:58 +0x8c
github.com/weaveworks/cortex/pkg/querier.(*nonOverlappingIterator).Next(0xc4206343c0, 0x74c766a8)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_merge_iterator.go:193 +0x140
github.com/weaveworks/cortex/pkg/querier.(*chunkMergeIterator).Next(0xc420108af0, 0x74c766a8)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_merge_iterator.go:112 +0x201
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage.(*BufferedSeriesIterator).Next(0xc4205f1e30, 0xc420c2ef01)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage/buffer.go:118 +0xea
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage.(*BufferedSeriesIterator).Seek(0xc4205f1e30, 0x74c7a140, 0x10)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/storage/buffer.go:100 +0xc4
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).matrixIterSlice(0xc42003a940, 0xc4205f1e30, 0x74c6b6e1, 0x74c7a140, 0xc420334b00, 0x1, 0x10, 0xc4205f1e00, 0x1, 0x1)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:1104 +0x1f6
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).eval(0xc42003a940, 0x2058960, 0xc420b90660, 0xc420c2f858, 0xc42003a940)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:870 +0x1a03
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*evaluator).Eval(0xc42003a940, 0x2058960, 0xc420b90660, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:614 +0x8e
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).execEvalStmt(0xc4205f1bc0, 0x2062c20, 0xc4200ec660, 0xc4200ec600, 0xc420108730, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:426 +0x696
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*Engine).exec(0xc4205f1bc0, 0x2062c20, 0xc4200ec660, 0xc4200ec600, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:340 +0x5a8
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql.(*query).Exec(0xc4200ec600, 0x2062be0, 0xc420104050, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/promql/engine.go:142 +0xde
github.com/weaveworks/cortex/pkg/querier.testQuery(0x2051c40, 0xc4201661e0, 0x204b160, 0xc420b90280, 0x9a7ec800, 0x1e30b31, 0xd, 0x2ac2140, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:140 +0x2bf
github.com/weaveworks/cortex/pkg/querier.TestChunkQueryable.func1(0xc4201661e0)
	/go/src/github.com/weaveworks/cortex/pkg/querier/chunk_queryable_test.go:88 +0x1af
testing.tRunner(0xc4201661e0, 0xc4207b6520)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 85 [chan receive]:
github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2aa1780)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0xac
created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.0
	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x231
FAIL	github.com/weaveworks/cortex/pkg/querier	60.789s
```

--- a/pkg/querier/chunk_queryable_test.go
+++ b/pkg/querier/chunk_queryable_test.go
@@ -83,7 +83,7 @@ func TestChunkQueryable(t *testing.T) {
 		for _, encoding := range encodings {
 			for _, query := range queries {
 				t.Run(fmt.Sprintf("%s/%s/%s", queryable.name, encoding.name, query.query), func(t *testing.T) {
-					store, from := makeMockChunkStore(t, 24*30, encoding.e)
+					store, from := makeMockChunkStore(t, 24*15, encoding.e)
 					queryable := queryable.f(store)
 					testQuery(t, queryable, from, query)
 				})

